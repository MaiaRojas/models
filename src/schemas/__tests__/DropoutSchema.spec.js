const mongoose = require('mongoose/browser');
const { DropoutSchema, CohortMembershipSchema } = require('..')(mongoose);

describe('DropoutSchema', () => {
  it('should fail validation when fields are missing', () => {
    const doc = new mongoose.Document({}, DropoutSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });
  it('should successfully validate with proper values', (done) => {
    const cohortMembership = new mongoose.Document({}, CohortMembershipSchema);
    const doc = new mongoose.Document({
      campus: 'lim',
      cohort: 'lim007',
      cohortMembership: cohortMembership._id,
      fullName: 'Diego Vélez',
      email: 'someone@somewhere.com',
      signUpCampus: 'BOG',
      date: '5/14/2018',
      stage: 1,
      project: 'Cipher',
      studentCode: 'LIM181080',
      reason: 'dropout',
      reasonDetail: 'Problemas familiares y económicos',
      notes: 'Una lástima, era una buena estudiante :(',
      isStaffSad: true,
      covidRelated: 'yes',
    }, DropoutSchema);
    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
