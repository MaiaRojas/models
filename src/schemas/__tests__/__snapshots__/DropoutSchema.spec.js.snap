// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropoutSchema should fail validation when fields are missing 1`] = `
Object {
  "campus": [ValidatorError: Path \`campus\` is required.],
  "cohort": [ValidatorError: Path \`cohort\` is required.],
  "covidRelated": [ValidatorError: Path \`covidRelated\` is required.],
  "email": [ValidatorError: Path \`email\` is required.],
  "fullName": [ValidatorError: Path \`fullName\` is required.],
  "notes": [ValidatorError: Path \`notes\` is required.],
  "reason": [ValidatorError: Path \`reason\` is required.],
  "reasonDetail": [ValidatorError: Path \`reasonDetail\` is required.],
  "signUpCampus": [ValidatorError: Path \`signUpCampus\` is required.],
  "staffSad": [ValidatorError: Path \`staffSad\` is required.],
  "stage": [ValidatorError: Path \`stage\` is required.],
  "studentCode": [ValidatorError: Path \`studentCode\` is required.],
}
`;
